apps/web/src/components/ResponsivePayloadImage/index.tsx

```diff
@@ -1,47 +1,81 @@
 'use client';
 
 import React from 'react';
-import type { ImageProps } from 'next/image';
 import Image from 'next/image';
+import get from 'lodash/get';
 import styled from 'styled-components';
 
 import genClassName from '@web/lib/genClassname';
-import type { Image as ImageT } from '@web/payload/payload-types';
-
-export type ResponsivePayloadImageType = {
-  image: ImageT;
-  imageProps?: Partial<ImageProps>;
-  className?: string;
-  classOverride?: string;
-};
+import type { PayloadImageT } from '@web/primitives/primitives';
 
 const ImageWrapper = styled.div`
   container-type: inline-size;
   position: relative;
+`;
 
-  img {
-    object-fit: cover;
+type Dimensions = {
+  height?: number;
+  width?: number;
+};
+/* If there is no height or width then the image will fill the container */
+const isFill = ({ width, height }: Dimensions): boolean => {
+  if (width === undefined && height === undefined) {
+    return true;
   }
-`;
+  return false;
+};
 
 function ResponsivePayloadImage({
-  image,
+  alt,
+  url,
+  height,
+  width,
   imageProps,
+  additionalProps,
   className,
   classOverride
-}: ResponsivePayloadImageType) {
-  const { alt, url: imageUrl } = image;
+}: PayloadImageT) {
+  /* Additional props for the image */
+  const fill = get(imageProps, 'fill', isFill({ height, width } as Dimensions));
```

Comment:
repeating the comment i made in slack, we should change this to use conditional chaining instead of the `get` method, with the hope that it's better at catching potential type errors. also one less dependency, and slightly less code.